cygnus-arcgis/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIArcgisFeatureTableSink.java

```diff
@@ -0,0 +1,789 @@
+                    String strValue = URLDecoder.decode(unquote(attrValue.toString()));
+                    feature.getAttributes().put(attrName, strValue);
+                } else {
+                    try {
+                        // Try to insert as Integer
+                        feature.addAttribute(attrName, Integer.parseInt(attrValue.toString()));
+                    } catch (NumberFormatException e2) {
+                        try {
+                            // Try to insert as Double
+                            feature.addAttribute(attrName, Double.parseDouble(attrValue.toString()));
+                        } catch (NumberFormatException e3) {
+                            // If all fails, insert as String
+                            LOGGER.warn(
+                                    "[NGSIArcgisAggregator] Unquoted String attribute: " + attrName + ":" + attrValue);
+                            String strValue = URLDecoder.decode(attrValue.toString());
+                            feature.addAttribute(attrName, strValue);
+                        }
+                    }
+                }
+
+        }
+
+    }
+
+    /**
+     * Try to convert Fiware DateTime String to Date object, If it can't, input string will be returned.
+     * 
+     * @param dateStr
+     * @return Date Object, or String.
+     */
+    private Object parseFiwareDate(String dateStr) {
+
+        // Normalize String date
+        dateStr = unquote(dateStr);
+        dateStr = dateStr.replaceAll("Z[\\s]*$", "+00:00");
+        dateStr = dateStr.replaceAll("([+-]{1}[0-9]{1,2})[\\s\\t\\r\\n]*$", "$1:00");
+
+        DateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+        try {
+            Date date = dateFormatter.parse(dateStr);
+            return date;
+        } catch (ParseException e) {
+            LOGGER.error("[NGSIArcgisAggregator] Unexpected DateTime format: " + dateStr);
+            return dateStr;
+        }
+    }
+
+    /**
+     * Determines if input sitring is quoted or not.
+     * 
+     * @param string
+     * @return boolean
+     */
+
+    private boolean isQuoted(String string) {
+        return string.matches("^[\\s]*\\\"(.*)\\\"[\\s]*$");
+    }
+
+    /**
+     * Unquote input string.
+     * 
+     * @param Unquoted
+     *            string
+     * @return
+     */
+    private String unquote(String string) {
+        return string.replaceAll("^[\\s]*\\\"(.*)\\\"[\\s]*$", "$1");
+    }
+
+    /**
+     * 
+     * @author PMO Santander Smart City – Ayuntamiento de Santander
+     *
+     */
+    public class ArcgisAggregatorDomain {
+        private CygnusLogger LOGGER = null;
+
+        // string containing the data aggregation
+        private Feature feature;
+
+        private String featureTableUrl;
+
+        private String uniqueField;
+
+        /**
+           * 
+           */
+        ArcgisAggregatorDomain(CygnusLogger logger) throws CygnusRuntimeError {
```

Comment:
Is it necesary that this class receives the logger object as a parametter? This inner class seems to be used only by the outer class, so it has acces to the outer class object. 

Would it be possible that this inner class usses the outer Logger object?

Reply:
Removed logger parameter.